esrally/metrics.py

```diff
@@ -254,8 +254,15 @@ def results_template(self):
     def _read(self, template_name):
         with open("%s/resources/%s.json" % (self.script_dir, template_name), encoding="utf-8") as f:
             template = json.load(f)
-            template["settings"]["index"]["number_of_shards"] = self._number_of_shards
-            template["settings"]["index"]["number_of_replicas"] = self._number_of_replicas
+            if self._number_of_shards is not None:
+                if int(self._number_of_shards) < 1:
+                    raise exceptions.SystemSetupError(
+                        f"The setting: datastore.number_of_shards must be >= 1. Please "
+                        f"check the configuration in {paths.rally_confdir()}/rally.ini"
```

Comment:
Note that due to undocumented parameter [configuration-name](https://github.com/elastic/rally/blob/aa1a89395783f712598d1a3bfd094a81ebeb1aa4/esrally/rally.py#L677) other ini files maybe references, e.g. in IT tests and elsewhere. Use `{self._config.config_file.location}` instead.

Reply:
I ended up changing this to point to regex (`.*/rally.ini`) instead, is that acceptable here?